server/models.py

```
@@ -100,18 +105,86 @@ class Tag(Base):
+
+    def get_messages(self, user1, user2):
+        try:
+            # Retrieve the chat history for both users (user1 and user2)
+            user = chat_collection.find_one({'username': user1})
+            if user:
+                for chatted_user in user.get('chatted_users', []):
+                    if chatted_user['username'] == user2:
+                        return chatted_user['chat_history']
+            return []
+        except Exception as e:
+            raise Exception(f"Error retrieving chat history: {str(e)}")
```

Ruff

Within an `except` clause, raise exceptions with `raise ... from err` or `raise ... from None` to distinguish them from errors in exception handling

(B904)